# Reference: 

# actual data source: https://stats.espncricinfo.com/ci/content/records/93276.html

**import required libraries**

In [2]:
import pandas as pd
import numpy as np

In [3]:
pip install openpyxl

Note: you may need to restart the kernel to use updated packages.


In [4]:
#for reading a excel file 
df = pd.read_excel("test_cricket.xlsx" , sheet_name='wickets')

# Naming the DataFrame - df
# Reading the .xlsx file using pandas: pd.read_excel
# Reading the sheet named 'wickets'

display(df.head(11))

#df.head(11) is for showing first 10 rows of the data frame.

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0   M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800   
1            SK Warne (AUS)  1992-2007  145   273  40705  17995   708   
2          A Kumble (INDIA)  1990-2008  132   236  40850  18355   619   
3         JM Anderson (ENG)  2003-2021  162   301  34791  16457   617   
4          GD McGrath (AUS)  1993-2007  124   243  29248  12186   563   
5           SCJ Broad (ENG)  2007-2021  148   272  29713  14502   523   
6             CA Walsh (WI)  1984-2001  132   242  30019  12688   519   
7             DW Steyn (SA)  2004-2019   93   171  18608  10077   439   
8       N Kapil Dev (INDIA)  1978-1994  131   227  27740  12867   434   
9         HMRKB Herath (SL)  1999-2018   93   170  25993  12157   433   
10       Sir RJ Hadlee (NZ)  1973-1990   86   150  21918   9611   431   

                    BBI                  BBM    Ave  Econ    SR   5  10  
0   1951-09-01 00:00:00               16/220  22.72  2.47  55.0  67  22  
1   1971-08-01 00:00:00               12/128  25.41  2.65  57.4  37  10  
2   1974-10-01 00:00:00               14/149  29.65  2.69  65.9  35   8  
3   1942-07-01 00:00:00  1971-11-01 00:00:00  26.67  2.83  56.3  30   3  
4   2021-08-24 00:00:00  2021-10-27 00:00:00  21.64  2.49  51.9  29   3  
5   2021-08-15 00:00:00               11/121  27.72  2.92  56.8  18   3  
6   1937-07-01 00:00:00                13/55  24.44  2.53  57.8  22   3  
7   1951-07-01 00:00:00  1960-11-01 00:00:00  22.95  3.24  42.3  26   5  
8   1983-09-01 00:00:00               11/146  29.64  2.78  63.9  23   2  
9                 9/127               14/184  28.07  2.80  60.0  34   9  
10  1952-09-01 00:00:00               15/123  22.29  2.63  50.8  36   9

**Observations:**

In this data set there are 14 columns .

 * player = player name
 

 * span = the staring year and the ending year of a player to play.
 

 * Mat = number of matches



* inns = number of innings



* balls = total number of bowling. a bowler delivers the ball from his or her end of the pitch towards the batsman



* Runs = total acquired runs in the whole  span



* Wkts= total number wicket taken of a bowler.



* BBI	 = BBI stands for Best Bowling in Innings and only gives the score for one innings. 



* BBM =  BBM stands for Best Bowling in Match and gives the combined score over 2 or more innings in one match. 



* average = bowling average is the number of runs they have conceded per wicket taken.



* Econ = Economy rate.  a player's economy rate is the average number of runs they have conceded per over bowled.



* SR	= strike rate. Bowling Strike Rate is a measurement of a bowler’s average number of balls bowled for every wicket 
taken. 



* 5	= number of times the player took 5 wickets in a match



 * 10 = number of times the player took 10 wickets in a match

In [5]:
# number of rows
print("number of rows = ", df.shape[0])

# number of columns
print("number of columns = ", df.shape[1] )

number of rows =  79
number of columns =  14


**Observations:**

The DataFrame contains 79 rows (players) and 14 columns (attributes).

In [6]:
df.describe()

#checking the data statistic 
#The describe () function is used to generate descriptive statistics 

Mat        Inns         Balls          Runs        Wkts  \
count   79.000000   79.000000     79.000000     79.000000   79.000000   
mean    80.101266  144.797468  18630.303797   8595.506329  317.101266   
std     28.537692   51.042310   7190.036515   3080.256645  121.731587   
min     37.000000   67.000000   8785.000000   4846.000000  200.000000   
25%     60.500000  110.000000  13580.000000   6456.500000  229.000000   
50%     71.000000  129.000000  16498.000000   7742.000000  266.000000   
75%     93.000000  169.000000  21742.500000   9756.000000  374.500000   
max    166.000000  301.000000  44039.000000  18355.000000  800.000000   

             Ave       Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  79.000000  
mean   27.466456   2.806582  59.187342  16.354430   2.797468  
std     3.657561   0.351666   9.349337   9.642372   3.235935  
min    20.940000   1.980000  41.200000   3.000000   0.000000  
25%    24.425000   2.600000  53.300000   9.500000   1.000000  
50%    28.000000   2.820000  57.400000  14.000000   2.000000  
75%    29.870000   3.080000  63.950000  20.500000   3.500000  
max    34.790000   3.460000  91.900000  67.000000  22.000000

# Descriptive Statistics:


shows the details of all numerical attributes in the dataset:

1. count = number of observations
2. mean = average of all values
3. std = standard deviation
4. min = minimum value among all the observations
5. 25% = the value at 25th percentile  
6. 50% = median value 
7. 75% = the value at 75th percentile.
8. max = maximum value among all the observations. 


**we can see that 25%,50%,75% values or observation has less than or falls below that point**


**Observations:**


1. The average number of matches played by the bowlers is 80 and minimum is 37 matches and maximum is 166. 25% bowler played approximately or less than 60 matches. 50% bowler played around or less than 71 matches, 75% bowler played nearly or less than 93 matches.


2. The average number of innings played by the bowlers is 144 and minimum is 67 ,maximum is 301. 25% bowler played approximately or less than 110 innings. 50% bowler played around or less than 129 innings, 75% bowler played nearly or less than 169 innings.



3. The average number of runs made by the bowlers is 8595. While the highest runs is 18355, only 25%  bowler scored less than 6456  runs.




4. The average number of wickets taken by the bowlers is 317, with 75% of bowlers scoring less than 375.



5. The average number of 5 wickets taken by the bowlers is 16 while 75 % bowlers scored less than 20.

In [7]:
print(df.info())

# checking for missing values and data types of each column

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     int64  
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(7), object(4)
memory usage: 8.8+ KB
None


**Observations:**


* we can see that there is no missing values present in the dataset

* there is 4 object data type (player, span, BBI, BBM)

* there is 3 float data type ( Strike Rate, Economy Rate, Average) and others are integer value.

# Renaming the columns

In [8]:
df = df.rename(columns={'Mat':'Matches', 
                        'Inns':'Innings',
                        'Wkts':'Wickets',
                        'Ave': 'Average',
                        'Econ' : 'Economy Rate',
                        'SR' : 'Strike Rate',
                        'BBI' : 'Best Bowling in Innings',
                        'BBM' : 'Best Bowling in Match',
                        5: '5_wickets',
                        10:'10_wickets'})


df.head()

Player       Span  Matches  Innings  Balls   Runs  \
0  M Muralitharan (ICC/SL)  1992-2010      133      230  44039  18180   
1           SK Warne (AUS)  1992-2007      145      273  40705  17995   
2         A Kumble (INDIA)  1990-2008      132      236  40850  18355   
3        JM Anderson (ENG)  2003-2021      162      301  34791  16457   
4         GD McGrath (AUS)  1993-2007      124      243  29248  12186   

   Wickets Best Bowling in Innings Best Bowling in Match  Average  \
0      800     1951-09-01 00:00:00                16/220    22.72   
1      708     1971-08-01 00:00:00                12/128    25.41   
2      619     1974-10-01 00:00:00                14/149    29.65   
3      617     1942-07-01 00:00:00   1971-11-01 00:00:00    26.67   
4      563     2021-08-24 00:00:00   2021-10-27 00:00:00    21.64   

   Economy Rate  Strike Rate  5_wickets  10_wickets  
0          2.47         55.0         67          22  
1          2.65         57.4         37          10  
2          2.69         65.9         35           8  
3          2.83         56.3         30           3  
4          2.49         51.9         29           3

**Observations:**

Muralitharan was the best bowling in innings in 1 septembe,1951 and total number best bowling in match was 16. 

His average rate is 22.72 ,economy rate was 2.47 and number of 5 wicktes taken in a match was 67.

In [9]:
df = df.drop("Best Bowling in Match", axis=1)

df.head()

Player       Span  Matches  Innings  Balls   Runs  \
0  M Muralitharan (ICC/SL)  1992-2010      133      230  44039  18180   
1           SK Warne (AUS)  1992-2007      145      273  40705  17995   
2         A Kumble (INDIA)  1990-2008      132      236  40850  18355   
3        JM Anderson (ENG)  2003-2021      162      301  34791  16457   
4         GD McGrath (AUS)  1993-2007      124      243  29248  12186   

   Wickets Best Bowling in Innings  Average  Economy Rate  Strike Rate  \
0      800     1951-09-01 00:00:00    22.72          2.47         55.0   
1      708     1971-08-01 00:00:00    25.41          2.65         57.4   
2      619     1974-10-01 00:00:00    29.65          2.69         65.9   
3      617     1942-07-01 00:00:00    26.67          2.83         56.3   
4      563     2021-08-24 00:00:00    21.64          2.49         51.9   

   5_wickets  10_wickets  
0         67          22  
1         37          10  
2         35           8  
3         30           3  
4         29           3